maven-core/src/main/java/org/apache/maven/graph/ProjectCollector.java

```diff
@@ -0,0 +1,37 @@
+package org.apache.maven.graph;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.ProjectBuildingException;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Facade to collect projects for a given set of pom.xml files.
+ */
+public interface ProjectCollector
+{
+    String HINT = "projectCollector";
```

Comment:
No need use hints anymore. If you see "hint", it was used by the Java 1.4  compatible plexus container. We now use @Inject / @Named

Reply:
Shouldn't this be called `ProjectsCollector`

Reply:
It should, and it now is.